engine/src/main/java/org/pentaho/di/kitchen/KitchenCommandExecutor.java

```diff
@@ -179,7 +179,7 @@ public Result execute( Params params ) throws Throwable {
     try {
 
       // Set the command line arguments on the job ...
-      job.setArguments( convert( KettleConstants.toJobMap( params ) ) );
+      job.setArguments( arguments );
```

Comment:
Seems like the older version of this had a check for when arguments size is 0 to be null. Not sure if it's required, but may be good to keep it:
`      if ( args.size() == 0 ) {
        job.setArguments( null );
      } else {
        job.setArguments( args.toArray( new String[args.size()] ) );
      }`

Reply:
I can see that part of code has changed on 8.1 from what you presented to this line: 
`job.setArguments( arguments != null ? arguments : null );`
Not sure if this line make sense as it is, mine does the same. So i'm guessing this is working this way since 8.1 even without that validation. Am i missing something here?

Reply:
`job.setArguments( arguments != null ? arguments : null );`
would make no sense, since that's the same as just
`job.setArguments( arguments );`

But `job.setArguments( args.size() != 0 ? args.toArray( new String[args.size()] ) : null );`, could make sense, since it's making the object passed being `null` if the args are empty. I'm just not sure if this validation is needed.

I'm ok keeping it as you have it: `job.setArguments( arguments );`, as long as it works well, both those checks seem unnecessary.

Reply:
Yes, i was pointing for that also, i can't understand that line!
But at the same time this tell us that size value validation is discarded since 8.1.